Enforce wordle rules and color letters in UI accordingly

Z1Epime opened the pull request:
- Added determination of the letter color for the UI + tests
- Changed input type from 1 TextBox per word to 1 TextBox per letter
- Letters in the UI are colored according to existence in the solution word

Z1Epime committed:
Determination of letter color in UI

MibFromGit commented on DataAccess/Model/Word.cs:
```diff
@@ -8,6 +8,8 @@ namespace DataAccess.Model
 {
     public class Word
     {
+        public const int WordLength = 5;
```
Das wurde schon gemerged

Z1Epime committed:
Added more Tests for rule enforcement

- Test for when a letter does not exists in a word and should return LetterHighlightColor.Gray
- Test for invalid input (does not run successfully at the moment)

Z1Epime committed:
Fixed RuleEnforcer and test for invalid input

- Fixed regex for determining non letters
- Renamed test
- Added german umlaute (ä, ö, ü) as test cases

SirPatagon commented on esWordle.Tests/RuleEnforcerTests.cs:
```diff
@@ -0,0 +1,330 @@
+                yield return new TestCaseData("R", 1, new Word("Knast"));
+                yield return new TestCaseData("R", 2, new Word("Knast"));
+                yield return new TestCaseData("R", 3, new Word("Knast"));
+                yield return new TestCaseData("R", 4, new Word("Knast"));
+
+                yield return new TestCaseData("U", 0, new Word("Knast"));
+                yield return new TestCaseData("U", 1, new Word("Knast"));
+                yield return new TestCaseData("U", 2, new Word("Knast"));
+                yield return new TestCaseData("U", 3, new Word("Knast"));
+                yield return new TestCaseData("U", 4, new Word("Knast"));
+
+                yield return new TestCaseData("V", 0, new Word("Knast"));
+                yield return new TestCaseData("V", 1, new Word("Knast"));
+                yield return new TestCaseData("V", 2, new Word("Knast"));
+                yield return new TestCaseData("V", 3, new Word("Knast"));
+                yield return new TestCaseData("V", 4, new Word("Knast"));
+
+                yield return new TestCaseData("W", 0, new Word("Knast"));
+                yield return new TestCaseData("W", 1, new Word("Knast"));
+                yield return new TestCaseData("W", 2, new Word("Knast"));
+                yield return new TestCaseData("W", 3, new Word("Knast"));
+                yield return new TestCaseData("W", 4, new Word("Knast"));
+
+                yield return new TestCaseData("X", 0, new Word("Knast"));
+                yield return new TestCaseData("X", 1, new Word("Knast"));
+                yield return new TestCaseData("X", 2, new Word("Knast"));
+                yield return new TestCaseData("X", 3, new Word("Knast"));
+                yield return new TestCaseData("X", 4, new Word("Knast"));
+
+                yield return new TestCaseData("Y", 0, new Word("Knast"));
+                yield return new TestCaseData("Y", 1, new Word("Knast"));
+                yield return new TestCaseData("Y", 2, new Word("Knast"));
+                yield return new TestCaseData("Y", 3, new Word("Knast"));
+                yield return new TestCaseData("Y", 4, new Word("Knast"));
+
+                yield return new TestCaseData("Z", 0, new Word("Knast"));
+                yield return new TestCaseData("Z", 1, new Word("Knast"));
+                yield return new TestCaseData("Z", 2, new Word("Knast"));
+                yield return new TestCaseData("Z", 3, new Word("Knast"));
+                yield return new TestCaseData("Z", 4, new Word("Knast"));
```
```suggestion
                var word = new Word("Knast");
                var alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ".Where(a => !word.Letters.ToUpperInvariant().Contains(a)); // Get alphabet without any letters of the word
                
                foreach (var letter in alphabet)
                {
                    for (int i = 0; i < 5; i++)
                        yield return new TestCaseData(letter.ToString(), i, wort);
                }
```

Z1Epime replied:
Applied

Z1Epime committed:
Changed way of accepting letter input

Input is now being accepted through 5 TextBoxes (1 per letter) instead of 1 per word.
This should make coloring single letters in the UI way easier.

Z1Epime committed:
Merge branch 'master' into rule-enforcement

Z1Epime committed:
Input TextBoxes now accept letters only

Z1Epime committed:
Color letters in UI according to existence

Z1Epime committed:
Refactored unit test

Z1Epime committed:
Wrapped latest test in region

Z1Epime commented:
Looks good 2 me 👍
